client/src/components/Problem/ProblemShow.js

```diff
@@ -98,29 +108,57 @@ class ProblemShow extends Component {
         push(`/courses/${courseId}/assignments/${assignment.get('id')}`);
     }
 
+    onAccept = accepted => {
+        const { dispatch, problem } = this.props;
+
+        this.setState({
+            rejected: null
+        })
+
+        dispatch(submitSubmission(this.getProblemId(), accepted))
+    }
+
+    onReject = rejected => {
+        this.setState({
+            rejected
+        })
+    }
+
     render() {
         const test = this.props.test;
         const assignment = this.getAssignment();
         const problem = this.getProblem(assignment);
+        const { user } = this.props;
```

Comment:
Combine this with `const test = this.props.test`, so `const { test, user } = this.props;`